src/chains/algorand_1/algoCompose.ts

```diff
@@ -48,18 +51,54 @@ const ComposeAction: { [key: string]: (args: any, suggestedParams: AlgorandTxHea
   Payment: PaymentTemplate,
 }
 
+/** compile an uncompiled TEAL program (into a Uint8Array) */
+export async function compileFromSourceCode(sourceCode: string, algoClient: AlgoClient): Promise<Uint8Array> {
+  const encoder = new TextEncoder()
+  const programBytes = encoder.encode(sourceCode)
+  const compileResponse = await algoClient.compile(programBytes).do()
+  return new Uint8Array(Buffer.from(compileResponse.result, 'base64'))
+}
+
+/** compile the TEAL program if needed */
+export async function compileIfSourceCodeAndEncode(
```

Comment:
This function name is misleading. Rename it to something clear like compileSourceCodeIfNeeded.
This function (and compileFromSourceCode) dont belong in compose - move to algorand_1/helpers/transactionHelpers.ts